components/locale_canonicalizer/README.md

```diff
@@ -1,6 +1,57 @@
 # icu_locale_canonicalizer [![crates.io](http://meritbadge.herokuapp.com/icu_locale_canonicalizer)](https://crates.io/crates/icu_locale_canonicalizer)
 
+[`icu_locale_canonicalizer`](crate) is one of the [`ICU4X`] components.
 
+This API provides functionality to canonicalize locale identifiers based
+upon [`CLDR`] data.
+
+It currently supports the minimize and maximize likely subtags algorithms
+as described in [`TR-35`].
+
+The maximize method potentially updates a passed in locale in place
+depending up the results of running the 'Add Likely Subtags' algorithm
+from [`TR-35`].
+
+This minimize method returns a new Locale that is the result of running the
+'Remove Likely Subtags' algorithm from [`TR-35`].
+
+## Examples
+
+```rust
+use icu_locale_canonicalizer::{CanonicalizationResult, LocaleCanonicalizer};
+use icu_locid::Locale;
+
+let provider = icu_testdata::get_provider();
+let lc = LocaleCanonicalizer::new(&provider).unwrap();
+
+let mut locale : Locale = "en-US".parse().unwrap();
```

Comment:
Nit: I prefer to use non-English locales in examples where it makes sense, both because it's more interesting and because it reflects our nature as an internationalization library.

An interesting example would be a pair of locales that have different scripts. Examples:

- zh-CN vs zh-TW (Hans vs Hant)
- sr-ME vs sr-BA (Latn vs Cyrl)